src/deprecated.jl

```diff
@@ -1,443 +1,3 @@
-                     "in `fs` have unique names.", :aggregate)
-    end
-    return df
-end
-
-function aggregate(gd::GroupedDataFrame, f::Any; sort::Bool=false)
-    df = combine(gd, valuecols(gd) .=> [f])
-    if sort
-        Base.depwarn("`aggregate(gd, f, sort=true)` is deprecated. Instead use" *
-                     " `df = combine(gd, valuecols(gd) .=> f); " *
-                     "sort!(df, Not(groupcols(gd)))`.",
-                     :aggregate)
-        sort!(df, names(df, Not(groupcols(gd))))
-    else
-        Base.depwarn("`aggregate(gd, f)` is deprecated. Instead use" *
-                     " `combine(gd, valuecols(gd) .=> f)`",
-                     :aggregate)
-    end
-    return df
-end
-
-function aggregate(gd::GroupedDataFrame, fs::AbstractVector; sort::Bool=false)
-    df = hcat([combine(gd, valuecols(gd) .=> [f], keepkeys=i==1) for (i, f) in enumerate(fs)]...,
-                       makeunique=true)
-    if sort
-        Base.depwarn("`aggregate(gd, fs, sort=true)` is deprecated. Instead use " *
-                     "`df = combine(gd, [names(gd) .=> f for f in fs]...); " *
-                     "sort!(df, names(df, Not(groupcols(gd))))`" *
-                     " if functions in `fs` have unique names.", :aggregate)
-        sort!(df, Not(groupcols(gd)))
-    else
-        Base.depwarn("`aggregate(gd, fs)` is deprecated. Instead" *
-                     " use `combine(gd, [names(gd) .=> f for f in fs]...)`" *
-                     " if functions in `fs` have unique names.", :aggregate)
-    end
-    return df
-end
-
-function aggregate(d::AbstractDataFrame, cols, f::Any;
-                   sort::Bool=false, skipmissing::Bool=false)
-    Base.depwarn("`aggregate(d, cols, f, sort=$sort, skipmissing=$skipmissing)` " *
-                 "is deprecated. Instead use " *
-                 "combine(groupby(d, cols, sort=$sort, skipmissing=$skipmissing)," *
-                 " names(d, Not(cols)) .=> f)`",
-                 :aggregate)
-    gd = groupby(d, cols, sort=sort, skipmissing=skipmissing)
-    df = combine(gd, valuecols(gd) .=> [f])
-    return df
-end
-
-function aggregate(d::AbstractDataFrame, cols, fs::AbstractVector;
-                   sort::Bool=false, skipmissing::Bool=false)
-    Base.depwarn("`aggregate(d, cols, fs, sort=$sort, skipmissing=$skipmissing)` " *
-                 " is deprecated. Instead use " *
-                 "combine(groupby(d, cols, sort=$sort, skipmissing=$skipmissing), "*
-                 "[names(d, Not(cols)) .=> f for f in fs]...)`" *
-                 " if functions in `fs` have unique names.", :aggregate)
-    gd = groupby(d, cols, sort=sort, skipmissing=skipmissing)
-    df = hcat([combine(gd, valuecols(gd) .=> [f], keepkeys=i==1) for (i, f) in enumerate(fs)]...,
-                       makeunique=true)
-    return df
-end
-
-@deprecate deleterows!(df::DataFrame, inds) delete!(df, inds)
-
-@deprecate by(f::Base.Callable, d::AbstractDataFrame, cols::Any;
```

Comment:
Could you keep a method for `by` pointing to the new syntax at least? It used the be the main function to perform grouping and it's been deprecated only recently. So it's likely to confuse many users who read old blog posts or forum answers. (`aggregate` is less essential, though I would have done the same.)

Reply:
OK - I will re-introduce `by` and `aggregate` but with a general error message (we have evolving rules of how we parse things, and these old blogs would not work anyway)